packages/aws_common/lib/src/io/aws_file_platform_html.dart

```diff
@@ -145,6 +142,38 @@ class AWSFilePlatform extends AWSFile {
         return blobType;
       });
 
+  @override
+  ChunkedStreamReader<int> getChunkedStreamReader() {
+    return ChunkedStreamReader(stream);
+  }
+
+  @override
+  Stream<List<int>> openRead([int? start, int? end]) {
+    if (super.isStreamOnly) {
```

Comment:
> Also technically you could implement this for any stream using the chunked stream reader

Yes, but it would a different behavior than `openRead` over a file, the latter allows in parallel random file access. 

If over a stream, e.g. `openRead(500, 600)`, it requires to listen to the stream, and ignore the first 499 events in order to get the 500th to start with, and after this,  `openRead(200, 300)` will be impossible correct?

Reply:
Gotcha- I didn't realize openRead was parallelized at the VM layer. You could use a `StreamSplitter` but it would buffer all bytes read to that point in order to allow random access.